src/System.Collections.Concurrent/tests/ProducerConsumerCollectionTests.cs

```diff
@@ -761,8 +768,11 @@ public void ManyConcurrentAddsTakes_CollectionRemainsConsistent(int operations)
                 // avoid 0 as default(T), to detect accidentally reading a default value
                 for (int i = 1; i < (1 + operations); i++)
                 {
-                    Assert.True(c.TryAdd(i));
-                    added.Add(i);
+                    if (c.Count < MaxConcurrentCollectionSize)
+                    {
+                        Assert.True(c.TryAdd(i));
+                        added.Add(i);
+                    }
```

Comment:
>How about just changing the test to have N threads all adding and taking, 

Cool, that's exactly what I'm changing the test to do now.  I didn't think about removing the count restrictions as well, though it makes sense.

If we always do an equal number of adds and takes, then I guess we can also remove the verification at the end and replace it with an assertion that the collection is empty.